src/proxy/RingbufferProxy.ts

```diff
@@ -0,0 +1,63 @@
+import * as Promise from 'bluebird';
+import {PartitionSpecificProxy} from './PartitionSpecificProxy';
+import {IRingbuffer} from './IRingbuffer';
+import {OverflowPolicy} from '../core/OverflowPolicy';
+import {RingbufferAddCodec} from '../codec/RingbufferAddCodec';
+import {RingbufferReadOneCodec} from '../codec/RingbufferReadOneCodec';
+import {RingbufferAddAllCodec} from '../codec/RingbufferAddAllCodec';
+import {RingbufferTailSequenceCodec} from '../codec/RingbufferTailSequenceCodec';
+import {RingbufferHeadSequenceCodec} from '../codec/RingbufferHeadSequenceCodec';
+import {RingbufferRemainingCapacityCodec} from '../codec/RingbufferRemainingCapacityCodec';
+import {RingbufferSizeCodec} from '../codec/RingbufferSizeCodec';
+import {RingbufferCapacityCodec} from '../codec/RingbufferCapacityCodec';
+import {RingbufferReadManyCodec} from '../codec/RingbufferReadManyCodec';
+import {Data} from '../serialization/Data';
+
+export class RingbufferProxy<E> extends PartitionSpecificProxy implements IRingbuffer<E> {
+
+    capacity(): Promise<Long> {
+        return this.encodeInvoke<Long>(RingbufferCapacityCodec);
+    }
+
+    size(): Promise<Long> {
+        return this.encodeInvoke<Long>(RingbufferSizeCodec);
+    }
+
+    tailSequence(): Promise<Long> {
+        return this.encodeInvoke<Long>(RingbufferTailSequenceCodec);
+    }
+
+    headSequence(): Promise<Long> {
+        return this.encodeInvoke<Long>(RingbufferHeadSequenceCodec);
+    }
+
+    remainingCapacity(): Promise<Long> {
+        return this.encodeInvoke<Long>(RingbufferRemainingCapacityCodec);
+    }
+
+    add(item: E, overflowPolicy: OverflowPolicy = OverflowPolicy.OVERWRITE): Promise<Long> {
```

Comment:
`overflowPolicy` is not specified as optional in `IRingBuffer`.
Also, Java RingBuffer's `addAsync` requires user to provide `overflowPolicy` everytime. I think there should not be a default value here.
Also, a null check is required for `overflowPolicy`